src/spicy/manager.cc

```diff
@@ -215,7 +215,7 @@ void Manager::registerPacketAnalyzer(const std::string& name, const std::string&
 void Manager::registerType(const std::string& id, const TypePtr& type) {
     auto [ns, local] = parseID(id);
 
-    if ( const auto& old = detail::lookup_ID(local.c_str(), ns.c_str()) ) {
+    if ( const auto& old = detail::lookup_ID(local.c_str(), ns.c_str(), true) ) {
```

Comment:
From the description:

> i.e., we didn't provide a no_global arg to detail::lookip_ID which defaulted to true

... defaulted to false? And so global lookup was true? because no_global=false is actually true?

> I'd argue that the behavior of that function is confusing and probably error-prone

TBH, last time I looked at the bool flags of this API I couldn't make much sense of the motivation. I think passing `same_module_only=true` in addition to your current fix might make the explanation easier even if it's redundant with `no_global=true`.

---
This should fix the problem, so I'm approving. If you add another `, true`, that's fine, too :-)

Reply:
Thanks for the careful reading, I'll fix the commit message to not add more confusion. I'll leave the args as is since I am fine with the defaults.